src/python/WMCore/MicroService/DataStructs/MSRuleCleanerWorkflow.py

```diff
@@ -0,0 +1,80 @@
+"""
+File       : MSRuleCleanerWflow.py
+Description: Provides a document Template for the MSRuleCleaner MicroServices
+"""
+
+
+from copy import deepcopy
+
+
+class MSRuleCleanerWflow(dict):
+    """
+    A minimal workflow and transfer information representation to serve the needs
+    of the MSRuleCleaner Micro Service.
+    """
+
+    def __init__(self, doc, **kwargs):
+        super(MSRuleCleanerWflow, self).__init__(**kwargs)
+
+        # Search for all the keys we need from the ReqManager workflow description
+        myDoc = {}
+        for tup in self.docSchema():
+            if tup[0] in doc:
+                myDoc[tup[0]] = deepcopy(doc[tup[0]])
+            else:
+                myDoc.update({tup[0]: tup[1]})
+        self.update(myDoc)
+
+    def docSchema(self):
+        """
+        Return the data schema for the document.
+        It's a tuple where:
+        * 1st element: is the key name / attribute in the request
+        * 2nd element: is the default value
+        * 3rd element: is the expected data type
+
+        Document format:
+            {
+            "RequestName": "ReqName",
+            "RequestType": "Type",
+            "RequestStatus": "Status",
+            "OutputDatasets": [],
+            'RulesToClean': {'plineMSTrCont': [],
+                             'plineMSTrBlock': [],
+                             'plineAgentCont': [],
+                             'plineAgentBlock': []},
+            'CleanupStatus': {'plineMSTrCont': False,
+                              'plineMSTrBlock': False,
+                              'plineAgentCont': False,
+                              'plineAgentBlock': False},
+            "TransferDone": False  # information - returned by the MSOutput REST call.
+            'TargetStatus': 'normal-archived' || 'rejected-achived' || 'aborted-archived',
+            'ParentageResolved': Bool,
+            'PlineMarkers': [None],
+            'IsClean': False
+            'ForceArchive', False]
+            }
+        :return: a list of tuples
+        """
+        docTemplate = [
+            ('RequestName', None, (str, unicode)),
+            ('RequestType', None, (str, unicode)),
+            ('RequestStatus', None, (str, unicode)),
+            ('OutputDatasets', [], list),
+            ('RulesToClean', {}, dict),
+            ('CleanupStatus', {}, dict),
+            ('TransferDone', False, bool),
+            ('TargetStatus', None, (str, unicode)),
+            ('ParentageResolved', True, bool),
+            ('PlineMarkers', None, list),
```

Comment:
Didn't we discuss that its default value would be an empty list?

Reply:
I remember I was keeping this because I wanted to have a way to distinguish between the cases when 'It is the very first pLine executed on that workflow' and  'No one have done any job on the workflow at all'. Since I have tested it with `None` value I'd rather keep it as it is. It will require a new set of workflows injected into the system so that I can test with a `[]` default.